### ISSUE TITLE
[rllib] slateq model does not work on Mac M1 since the dependent simulator recsim

### ISSUE THREAD

**Issue by junwucs** [ray-project/ray#26181]
### What happened + What you expected to happen

pip install recsim==0.2.4
Collecting recsim==0.2.4
  Using cached recsim-0.2.4.tar.gz (65 kB)
  Preparing metadata (setup.py) ... done
Requirement already satisfied: absl-py in /Users/xxx/miniforge3/envs/yyy/lib/python3.9/site-packages (from recsim==0.2.4) (1.1.0)
Collecting dopamine-rl>=2.0.5
  Using cached dopamine_rl-4.0.5-py3-none-any.whl (174 kB)
Collecting gin-config
  Using cached gin_config-0.5.0-py3-none-any.whl (61 kB)
Requirement already satisfied: gym in /Users/xxx/miniforge3/envs/yyy/lib/python3.9/site-packages (from recsim==0.2.4) (0.21.0)
Requirement already satisfied: numpy in /Users/xxx/miniforge3/envs/yyy/lib/python3.9/site-packages (from recsim==0.2.4) (1.22.4)
Requirement already satisfied: scipy in /Users/xxx/miniforge3/envs/yyy/lib/python3.9/site-packages (from recsim==0.2.4) (1.8.1)
ERROR: Could not find a version that satisfies the requirement tensorflow (from recsim) (from versions: none)
ERROR: No matching distribution found for tensorflow

### Versions / Dependencies

macOS Monterey 12.4 # **Apple M1 Max**
python=3.9.13
ray >= 1.13.0
gym>=0.21.0,<0.24.0; python_version >= '3.7'
recsim==0.2.4
tensorflow-macos # **mac tensorflow version:** https://developer.apple.com/metal/tensorflow-plugin/

### Reproduction script

It went wrong each time and I could not walk through. I believe it caused by the dependent package **recsim**, and this package depends on TensorFlow with a version not supported by Mac M1  yet.  Can anyone help me work out? 

Thanks, really.

### Issue Severity

High: It blocks me from completing my task.

**Comment by junwucs** [comment]
Anyone can help me out? Thanks~

**Comment by ArturNiederfahrenhorst** [comment]
Hi @junwucs ,

I have struggled in the past with getting recsim to run on my M1. **This is not an RLlib specific problem though.**
I've now resorted to installing it with `pip install --no-deps recsim` and installing dependencies that pop up as missing when importing recsim by hand. This worked out and I am running RLlib and Recsim together.
Other developers run it together with RLlib as well.
